cmd/snap/cmd_services.go

```diff
@@ -117,12 +125,70 @@ func svcNames(s []serviceName) []string {
 	return svcNames
 }
 
+func fmtServiceStatus(svc *client.AppInfo, isGlobal bool) string {
+	startup := i18n.G("disabled")
+	if svc.Enabled {
+		startup = i18n.G("enabled")
+	}
+
+	// When requesting global service status, we don't have any active
+	// information available for user daemons.
+	current := i18n.G("inactive")
+	if svc.DaemonScope == snap.UserDaemon && isGlobal {
+		current = "-"
+	} else if svc.Active {
+		current = i18n.G("active")
+	}
+
+	return fmt.Sprintf("%s.%s\t%s\t%s\t%s", svc.Snap, svc.Name, startup, current, clientutil.ClientAppInfoNotes(svc))
+}
+
+func (s *svcStatus) showGlobalEnablement(u *user.User) bool {
+	if u.Uid == "0" && !s.User {
+		return true
+	} else if u.Uid != "0" && s.Global {
+		return true
+	}
+	return false
+}
+
+func (s *svcStatus) validateArguments(u *user.User) error {
+	// can't use --global and --user together
+	if s.Global && s.User {
+		return fmt.Errorf(i18n.G("cannot combine --global and --user switches."))
+	}
+
+	// --user is supported for root
+	if s.User && u.Uid != "0" {
+		return fmt.Errorf(i18n.G("unsupported argument --user when not root."))
+	}
+
+	// --global is supported for non-root
+	if s.Global && u.Uid == "0" {
+		return fmt.Errorf(i18n.G("unsupported argument --global when root."))
+	}
```

Comment:
what's the reason to error instead of ignoring the option that asks for the default again?

Reply:
I can change it to be a warning instead maybe? I still feel like educating the user is fine when they do something that's not needed.

Reply:
I see no precedence for this anywhere, I'll remove it